genomedata/__init__.py

```diff
@@ -264,6 +238,14 @@ def __del__(self):
         if self.isopen:
             self.close()
 
+    def _create_chromosome(self, name, mode):
+        name = "_".join(name.split())  # Remove any whitespace
+
+        res = self._chromosomes.create(name)
+        res.attrs.dirty = True
```

Comment:
`.attrs.dirty` is HDF5-specific, this should be moved in the ChromosomeList object

Reply:
There are some "dirty" checks throughout. I've replaced `attrs` with a SimpleNamespace object by default for non-persistant storage cases. I can still try to make it a clean break and put HDF5 only though.

Reply:
The reason why this is currently kept is `attrs` is used as a property to store metadata attributes about the chromosome, and the genome by proxy. It is also technically it is possible through the public interface to use something like `chr1.attrs.min`, so I wanted to keep this interface available. Maybe it ideally makes sense to make `dirty` accessible differently since it's not meant really to be used a public interface?

Reply:
We should not have `Chromosome.attrs` as a public interface. There is `Chromosome.min`, right?

Reply:
Yes, it is actually implemented as `return self.attrs.mins`. I'll work at putting `attrs` as an hdf5 only private property.